airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+import logging
+from typing import Any, List, Mapping, Tuple
+
+from airbyte_cdk.models.airbyte_protocol import SyncMode
+from airbyte_cdk.sources.declarative.checks.connection_checker import ConnectionChecker
+from airbyte_cdk.sources.source import Source
+
+
+class CheckStream(ConnectionChecker):
+    """
+    Checks the connections by trying to read records from one or many of the streams
+    """
+
+    def __init__(self, stream_names: List[str]):
+        self._stream_names = set(stream_names)
+
+    def check_connection(self, source: Source, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        for stream in source.streams(config):
+            if stream.name in self._stream_names:
```

Comment:
should we fail if a stream the developer chose for connection check doesn't exist in the catalog?

Reply:
@sherifnada wouldn't this make the `check` function dependent on the `discover`?

Reply:
it doesn't make it more dependent than the current state since we're already getting the `streams` object